credentials/java/com/google/auth/Credentials.java

```diff
@@ -54,6 +56,19 @@ public abstract class Credentials implements Serializable {
    */
   public abstract String getAuthenticationType();
 
+  /**
+   * Returns the universe domain for the credential.
+   *
+   * @return a universe domain value in the format some-domain.xyz. By default, returns the Google
+   *     universe domain googleapis.com.
+   * @throws IOException extending classes might have to do remote calls to determine the universe
+   *     domain. The exception should implement {@link Retryable} and {@code isRetryable()} will
```

Comment:
```suggestion
   *     domain. The exception implements {@link Retryable} and {@code isRetryable()} will
```

nit: Correct me if I'm wrong but for `getUniverseDomain()` the IOException will always be a Retryable interface, right? Credentials without a remote call shouldn't throw an IOException.

Reply:
For credentials we control - yes. Otherwise - we cannot guarantee. We can change it for "must implement".